File: objc/Demo Objc/CapturePanViewController.m
//
//  CapturePanViewController.m
//  Demo Objc
//
//  Created by Eduardo Mello de Macedo | Stone on 13/03/17.
//  Copyright © 2017 Eduardo Mello de Macedo | Stone. All rights reserved.
//

#import "CapturePanViewController.h"

@interface CapturePanViewController ()
@property (weak, nonatomic) IBOutlet UILabel *feedback;

@end

@implementation CapturePanViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.navigationItem.title = @"Captura de PAN";
    
    self.overlayView = [[UIView alloc] initWithFrame:[UIScreen mainScreen].bounds];
    self.overlayView.backgroundColor = [UIColor colorWithRed:0 green:0 blue:0 alpha:0.5];
    self.activityIndicator = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleWhiteLarge];
    self.activityIndicator.center = self.overlayView.center;
    
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)performCapture:(id)sender {
    
    [self.overlayView addSubview:self.activityIndicator];
    [self.activityIndicator startAnimating];
    [self.navigationController.view addSubview:self.overlayView];
    
    [STNCardProvider getCardPan:^(BOOL succeeded, NSString *pan, NSError *error) {
        [self.overlayView removeFromSuperview];
        // verifica se a requisição ocorreu com sucesso
        if (succeeded) {
            NSLog(@"**** **** **** %@", pan);
            self.feedback.text =  [NSString stringWithFormat:@"Os 4 ultimos digitos são: %@", pan];
        } else {
            NSLog(@"%@", error.description);
            self.feedback.text = error.description;
        }
    }];
}

@end

File: swift/Demo Swift/Demo Swift/CapturePanViewController.swift
//
//  CapturePanViewController.swift
//  Demo Swift
//
//  Created by Eduardo Mello de Macedo | Stone on 21/03/17.
//  Copyright © 2017 Eduardo Mello de Macedo | Stone. All rights reserved.
//

import UIKit
import StoneSDK

class CapturePanViewController: UIViewController {

    @IBOutlet weak var feedbackLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.navigationItem.title = "Captura de PAN"
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    @IBAction func performCapture(_ sender: Any) {
        STNCardProvider.getCardPan { (succeeded, pan, error) in
            if (succeeded) {
                NSLog("**** **** **** \(pan)")
                self.feedbackLabel.text = "Os 4 ultimos digitos são:\(pan)"
            } else {
                NSLog(error.debugDescription)
                self.feedbackLabel.text = error.debugDescription
            }
        }
    }
}
